src/unix_term.rs

```diff
@@ -66,41 +65,76 @@ pub(crate) fn terminal_size(out: &Term) -> Option<(u16, u16)> {
     }
 }
 
-pub(crate) fn read_secure() -> io::Result<String> {
-    let mut f_tty;
-    let fd = unsafe {
-        if libc::isatty(libc::STDIN_FILENO) == 1 {
-            f_tty = None;
-            libc::STDIN_FILENO
+enum Input<T> {
+    Stdin(io::Stdin),
+    File(T),
+}
+
+impl Input<fs::File> {
+    fn new() -> io::Result<Self> {
+        let stdin = io::stdin();
+        if is_a_terminal(&stdin) {
+            Ok(Input::Stdin(stdin))
         } else {
             let f = fs::OpenOptions::new()
                 .read(true)
                 .write(true)
                 .open("/dev/tty")?;
-            let fd = f.as_raw_fd();
-            f_tty = Some(BufReader::new(f));
-            fd
+            Ok(Input::File(f))
```

Comment:
As suggested previously, please inline the `f` binding.

Reply:
Indeed that was suggested earlier, but I find that harder to read because of the error handling in the `f` expression. I'd prefer not to have `?` in the middle of an expression.